server/services/ticket.js

```diff
@@ -0,0 +1,64 @@
+import Joi from 'joi';
+import Ticket from '../models/ticket';
+
+const ticketSchema = {
+  groupId: Joi.string().required(),
+  metadata: Joi.object().required()
+};
+
+const idPattern = Joi.string().hex().length(24).required();
+
+function getById(id) {
+  return new Promise((resolve, reject) => {
+    Joi.validate(id, idPattern, (err, value) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(Ticket.findById(value));
+      }
+    });
+  });
+}
+
+function createTicket(ticketDetails) {
+  return new Promise((resolve, reject) => {
+    Joi.validate(ticketDetails, ticketSchema, (err, value) => {
+      if (err) {
```

Comment:
I'm not sure if I entirely agree with removing parens here. Is this a javascript convention that is widely used? If so could you link it @paradime?